Public/Src/Cache/ContentStore/Distributed/NuCache/RocksDbContentLocationDatabase.cs

```diff
@@ -197,9 +196,14 @@ private BoolResult Load(OperationContext context, StoreSlot activeSlot, bool cle
                 var possibleStore = KeyValueStoreAccessor.Open(storeLocation,
                     additionalColumns: new[] { nameof(Columns.ClusterState), nameof(Columns.Metadata) },
                     rotateLogs: true,
-                    failureHandler: failure =>
+                    failureHandler: failureEvent =>
                     {
-                        Tracer.Error(context, $"RocksDb critical error caused store deprecation: {failure.DescribeIncludingInnerFailures()}");
+                        bool isUserError = failureEvent.Rethrow;
+                        failureEvent.Invalidate = !isUserError;
+                        failureEvent.Rethrow = isUserError;

```

Comment:
2 nits:

1. Rethrow is set to itself:)
2. Does it make sense to have a helper method on the failureEvent instance, something: "InvalidateOnUserError" that will set the right flags?

Reply:
That would be weird, because it would effectively have to operate on the flag's current state, which may or may not match what is expected if the user changed it. Hence,

```
failureEvent.InvalidateOnUserError()
failureEvent.InvalidateOnUserError()
```

Would be strictly wrong (because it wouldn't be idempotent). I'd rather leave it as-is and added a comment.